src/sentry/search/utils.py

```diff
@@ -468,6 +468,40 @@ def _run_latest_release_query(
     return [row[0] for row in cursor.fetchall()]
 
 
+def get_terminal_release_for_group(
+    group: Group,
+    query_type: LatestReleaseOrders,
+    latest: bool,
+) -> Sequence[str]:
+    direction = "DESC" if latest else "ASC"
+    extra_conditions = ""
+    if query_type == LatestReleaseOrders.SEMVER:
+        rank_order_by = f"major {direction}, minor {direction}, patch {direction}, revision {direction}, CASE WHEN (prerelease = '') THEN 1 ELSE 0 END {direction}, prerelease {direction}, sr.id {direction}"
+        extra_conditions += " AND sr.major IS NOT NULL"
+    else:
+        rank_order_by = f"COALESCE(date_released, date_added) {direction}"
+
+    query = f"""
+        SELECT sr.*
+        FROM sentry_release sr
+        INNER JOIN (
+            SELECT sgr.release_id
+            FROM sentry_grouprelease sgr
+            WHERE sgr.group_id = %s
+            ORDER BY sgr.release_id {direction}
```

Comment:
I checked that we have an index for this column
https://github.com/getsentry/sentry/blob/2fef7faed341554a5f1b5e26423ea1ef2212079f/src/sentry/models/grouprelease.py#L26

I hope that mitigates perf concerns for an order-by.

Reply:
Hmm, yeah we might want to add an index on `group_id, release_id` here, good call. Since we're not releasing this broadly at first, I'm ok with merging this as is and following up with the index afterwards. The index will only matter in cases where customers have ridiculous numbers of releases (like 100k+) associated with a group, which is mostly a misuse of the system.

Reply:
Actually, there's an index on `group_id, first_seen` already. I'll just use that - it has the same effect